contrib/android/buildozer_qml.spec

```diff
@@ -84,7 +85,7 @@ android.permissions = INTERNET, CAMERA, WRITE_EXTERNAL_STORAGE
 
 # (int) Android API to use  (compileSdkVersion)
 # note: when changing, Dockerfile also needs to be changed to install corresponding build tools
-android.api = 30
+android.api = 31
 
 # (int) Android targetSdkVersion
 android.target_sdk_version = 31
```

Comment:
Note: the google play store requires target api 33 now.
So ideally we should set `android.api = 33` and comment out `android.target_sdk_version` (The latter was added as a crutch last year as we could not bump `android.api` to 31, which was the new minimum at the time. see https://github.com/spesmilo/electrum/commit/0efc8815a4431e000d86779bb49a360f6f486726 and https://github.com/spesmilo/electrum/pull/8010)

However, we might have to bump the qt version to 6.5 for this. :/

I've downloaded [different NDKs](https://github.com/android/ndk/wiki/Unsupported-Downloads), and it looks like
23(x) that this PR sets and that Qt 6.2-6.4 requires only supports compileSdkVersion up to 31.
25x is the first NDK that supports compileSdkVersion 33.
qt 6.5 is the first version that [uses NDK 25](https://doc.qt.io/qt-6.5/android-getting-started.html#android-ndk).

Alternatively, we might be able to get away with the same trick as last year and set targetSdkVersion higher than compileSdkVersion. I think whether that works depends on the exact changes we have to make to accommodate the runtime checks enforced on the app due to the raised targetSdkVersion.

----

Nevertheless, we can do this in a follow-up to the PR. Upgrading to *some version* of Qt6 is already a huge step towards this.

Reply:
I hope we can avoid Qt6.5, as this version has changes in the styling of the controls, which would force us to tidy up a lot of UI (probably not that difficult, just a big diff)

Reply:
As my main testing device was API33 for months and now recently API34, I think we can safely declare compatibility with targetSdkVersion=33

Reply:
> As my main testing device was API33 for months and now recently API34, I think we can safely declare compatibility with targetSdkVersion=33

AFAIU it should not matter what device you are testing on, as long as it has API >= targetSdkVersion.
As in, the app when running on your device behaves as per `min(targetSdkVersion, installed_android_on_device)`.

When bumping the targetSdkVersion, we should read through the documented changes at https://developer.android.com/about/versions/13/behavior-changes-13 and see if we need to adapt anything. I did that just now and it looks to me we should not be affected.

When we [bumped to 31](https://developer.android.com/google/play/requirements/target-sdk#pre13),  I also read through the corresponding changes, and had to backport some changes from upstream p4a. (e.g. https://github.com/sombernight/python-for-android/commit/b475b97b08e8a64b5d978b971d47a5e81d82d66c)  I even missed something for the notifications, as I think they were not really used for kivy (I think there was some code but disabled): https://github.com/kivy/plyer/pull/756

Anyway, I think we might be ok and can just bump it. :)

Reply:
> AFAIU it should not matter what device you are testing on, as long as it has API >= targetSdkVersion. As in, the app when running on your device behaves as per `min(targetSdkVersion, installed_android_on_device)`.

It does matter what API level you are testing against, precisely because of the behavior changes you already mentioned above, and any further restrictions added by newer API levels. `targetSdkVersion` merely suggests that the package is tested against that level.

On the other end of the spectrum, if a newer API level requires a change, it might need additional code to remain 
[backward compatible](https://developer.android.com/develop/ui/views/search/training/backward-compat) or else bumping the `minSdkVersion` in the manifest.

So, ideally, we need to test against ~every~ API level :)

We are fortunately not using bleeding edge API features, or parts of the API that are moving targets, so we can mostly assume things to keep working (taking the published android version changes into account of course)

Reply:
> targetSdkVersion merely suggests that the package is tested against that level.

Another way of saying what I meant is that for the testing to count in the first place, you have to test while already having set the higher targetSdkVersion.
Unless I completely misunderstood how it works :D

Reply:
tested with compileSdk 31, targetSdk 33, on SDK 34 device.
We should be good.